### PR TITLE
allow configuring topologySpreadConstraints in Helm chart

### LINKED ISSUE #2629: Allow configuring topologySpreadConstraints in Helm chart
Documentation: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/

`topologySpreadConstraints`, unlike pod anti-affinity, allow even distribution among of pods among a topology domain, e.g. availability zones.

Configuring them in the chart is not possible at this time.

### PR DESCRIPTION
**Pull request by kdomanski**
### Proposed changes
This allows a user of the Helm chart to add topologySpreadConstraints to the ingress pots, for example in order to ensure even distribution among availability zones.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

Fixes #2629

### CODE REVIEW

**Comment by brianehlert** [comment]
For reviewers: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/

**Comment by brianehlert** [comment]
Thanks @kdomanski 
Could you please open a corresponding issue to help us track this PR?

**Comment by kdomanski** [comment]
[Done](https://github.com/nginxinc/kubernetes-ingress/issues/2629)

File: deployments/helm-chart/README.md
Comment (kdomanski) [reply deployments/helm-chart/README.md]: Done

**Comment by kdomanski** [comment]
> Would you mind adding the field for the `DaemonSet` as well?

Done.

File: deployments/helm-chart/values.yaml
Comment (lucacome) [inline deployments/helm-chart/values.yaml:121]:
```suggestion
  ## The topology spread constraints of the Ingress Controller pods.
```

**Review by lucacome — APPROVED** [review]
LGTM

File: deployments/helm-chart/values.yaml
Comment (kdomanski) [reply deployments/helm-chart/values.yaml]: Fixed

**Comment by kdomanski** [comment]
@ciarams87 Hey, can I get a second review from you?

**Comment by lucacome** [comment]
@kdomanski can you rebase your branch with `main`?

**Pull request merged**
